snark/src/lib.rs

```diff
@@ -101,30 +93,53 @@ pub enum UniversalSetupIndexError<Bound, E> {
 
 /// A SNARK with universal setup. That is, a SNARK where the trusted setup is
 /// circuit-independent.
-pub trait UniversalSetupSNARK<F: PrimeField>: SNARK<F> {
+pub trait UniversalSetupSNARK<R: NPRelation>: SNARK<R> {
     /// Specifies how to bound the size of public parameters required to
     /// generate the index proving and verification keys for a given
     /// circuit.
```

Comment:
Lets add an example of what an index bound would be for a particular universal setup?

Reply:
Done.